### PR TITLE
ignition: add butane link

### PR DESCRIPTION
**Pull request by tormath1**
Signed-off-by: Mathieu Tortuyaux <mtortuyaux@microsoft.com>

<hr>

### CODE REVIEW

File: docs/provisioning/ignition/specification.md [hunk 1]
```diff
@@ -4,17 +4,16 @@ linktitle: Specification
...
-[ct-config]: ../config-transpiler/configuration
 
 ## Ignition v3
 
 Starting from release 3185.0.0, Ignition v3 (specification 3.3.0) is supported in addition of Ignition v2. There are some things to be aware of:
 * v1 and v2 are still supported and get translated at runtime; while this is tested well there may be corner cases where the v2 config relied on unspecified behavior
 * `clevis` is not supported
 * `kernelArguments` are supported and will persist the changes in `/usr/share/oem/grub.cfg` before the reboot but it only works for unconditional `set linux_append` statements in grub.cfg and `linux_console` is not considered
-* `ct` can't emit Ignition v3 yet and Butane's variant is not yet defined
+* [butane][butane-spec] can be used to generate Ignition v3 specification
```
Comment (pothos) [inline docs/provisioning/ignition/specification.md:16]:
```suggestion
* [butane][butane-spec] can be used to generate Ignition v3 configs
```

File: docs/provisioning/ignition/specification.md [hunk 2]
```diff
@@ -190,7 +189,7 @@ Starting from release 3185.0.0, Ignition v3 (specification 3.3.0) is supported i
 
 ## Ignition v2
 
-Ignition v2 is not developed anymore but still supported (specification 2.3.0):
+Ignition v2 is not developed anymore but still supported (specification 2.3.0), [ct][ct-config] can be used to emit Ignition v2 spec:
```
Comment (pothos) [inline docs/provisioning/ignition/specification.md:192]:
```suggestion
Ignition v2 is not developed anymore but still supported (specification 2.3.0), the high-level [Container Linux Config YAML format][ct-config] can be used to emit Ignition v2 configs:
```

File: docs/provisioning/ignition/specification.md [hunk 3]
```diff
@@ -4,17 +4,16 @@ linktitle: Specification
 weight: 30
 ---
 
-Ignition uses a JSON format that is specified in several versions. The traditional specification version 2 was used in CoreOS Container Linux and is also used in Flatcar Container Linux. The new specification version 3 has breaking changes and is used in Fedora CoreOS but also soon available in Flatcar Container Linux. The high-level Container Linux Config YAML format specification can be found [here][ct-config].
+Ignition uses a JSON format that is specified in several versions: v1, v2 and v3. While v1 and v2 are still supported in Flatcar Container Linux, it's recommended to write new configuration with v3.
```
[the remark this answers is not in the document]
Comment (tormath1) [reply docs/provisioning/ignition/specification.md:7]: so this is something I was thinking about actually: documentation does not seem to follow the Flatcar's release cycle - served documentation is always the reflect of the current stable release.
User who's using alpha / beta / LTS releases has no way to use the documentation associated to a given release...
We could think about having the following links:
- https://www.flatcar.org/docs/latest/: stable
- https://www.flatcar.org/docs/alpha/: alpha
- https://www.flatcar.org/docs/beta/: beta
...

[the remark this answers is not in the document]
Comment (pothos) [reply docs/provisioning/ignition/specification.md:7]: The more meaningful way would be by major version, not by channel, I would say.
Due to the overhead this wasn't done.

I think it's good to state when a feature is available.
```suggestion
Ignition uses a JSON format that is specified in several versions: v1, v2 and v3. While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.
```

File: docs/provisioning/ignition/specification.md [hunk 4]
```diff
@@ -4,17 +4,34 @@ linktitle: Specification
 weight: 30
 ---
 
-Ignition uses a JSON format that is specified in several versions. The traditional specification version 2 was used in CoreOS Container Linux and is also used in Flatcar Container Linux. The new specification version 3 has breaking changes and is used in Fedora CoreOS but also soon available in Flatcar Container Linux. The high-level Container Linux Config YAML format specification can be found [here][ct-config].
+Ignition uses a JSON format that is specified in several versions: v1, v2 and v3. While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.
```
Comment (pothos) [inline docs/provisioning/ignition/specification.md:7]:
```suggestion
Ignition uses a JSON format that is specified in several versions: v1, v2 and v3 (which itself have minor versions like `2.3.0`). While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.
```

**Review by pothos — APPROVED** [review]

File: docs/provisioning/ignition/specification.md [hunk 5]
```diff
@@ -4,17 +4,34 @@ linktitle: Specification
 weight: 30
 ---
 
-Ignition uses a JSON format that is specified in several versions. The traditional specification version 2 was used in CoreOS Container Linux and is also used in Flatcar Container Linux. The new specification version 3 has breaking changes and is used in Fedora CoreOS but also soon available in Flatcar Container Linux. The high-level Container Linux Config YAML format specification can be found [here][ct-config].
+Ignition uses a JSON format that is specified in several major versions: v1, v2 and v3 (which itself has minor versions like `2.3.0`). While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.
```
Comment (krnowak) [inline docs/provisioning/ignition/specification.md:7]: `v3` has minor versions like `2.3.0`? Or is it `v2`?

Comment (pothos) [reply docs/provisioning/ignition/specification.md:7]: v2 and v3 just refer to the major version, like 2.3.0 and 3.3.0

**Pull request merged**